control/summary_table.go

```diff
@@ -71,27 +73,104 @@ func (st *SummaryTable) Fill(openControlData opencontrols.Data) (err error) {
 	return
 }
 
+// diffControlOrigination computes the diff of the control origination.
+func (st *SummaryTable) diffControlOrigination(control string,
+	openControlData opencontrols.Data) ([]reporter.Reporter, error) {
+	// find the control origination section in the document.
+	docControlOriginationData, err := newControlOrigination(st)
+	if err != nil {
+		return nil, err
+	}
+	// find the control origins currently checked in the section in the doc.
+	docControlOrigins := docControlOriginationData.getCheckedOrigins()
+
+	// find the control origins noted in the yaml.
+	yamlControlOriginationData := openControlData.GetControlOrigins(control)
+	// find the control origins currently checked in the section in the YAML.
+	yamlControlOrigins := getCheckedOriginsFromYAML(yamlControlOriginationData)
+
+	// find the difference of the two sets.
+	controlOriginMap := getControlOriginMappings()
+	reports := []reporter.Reporter{}
+
+	// find only the origins in the document.
+	onlyInDocOrigins := set.Difference(docControlOrigins, yamlControlOrigins)
+	// create the diff report for the origins only in the document.
+	onlyInDocOriginReports := createControlOriginsDiffReport(onlyInDocOrigins, controlOriginMap, control, sspSrc)
+	reports = append(reports, onlyInDocOriginReports...)
+
+	// find only the origins in the yaml.
+	onlyInYAMLOrigins := set.Difference(yamlControlOrigins, docControlOrigins)
+	// create the diff report for the origins only in the yaml.
+	onlyInYAMLOriginReports := createControlOriginsDiffReport(onlyInYAMLOrigins, controlOriginMap, control, yamlSrc)
+	reports = append(reports, onlyInYAMLOriginReports...)
```

Comment:
i'll leave this as-is. i could move the Difference line inside `createControlOriginsDiffReport` but you would lose variable names like `onlyInDocOrigins` and `onlyInYAMLOrigins` and it would make it harder to follow